core/ecschema-editing/src/Editing/Properties.ts

```diff
@@ -0,0 +1,271 @@
+    derivedProperties.forEach((prop: MutableProperty) => {
+      prop.setName(newName);
+    });
+
+    existingProperty.setName(newName);
+  }
+
+  /**
+   * Sets the property description.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param description The new description to set.
+   */
+  public async setDescription(classKey: SchemaItemKey, propertyName: string, description: string) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setDescription(description);
+  }
+
+  /**
+   * Sets the property label.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param label The new label to set.
+   */
+  public async setLabel(classKey: SchemaItemKey, propertyName: string, label: string) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setLabel(label);
+  }
+
+  /**
+   * Sets the property isReadOnly attribute.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param isReadOnly The new isReadOnly value.
+   */
+  public async setIsReadOnly(classKey: SchemaItemKey, propertyName: string, isReadOnly: boolean) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setIsReadOnly(isReadOnly);
+  }
+
+  /**
+   * Sets the property priority.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param priority The new priority.
+   */
+  public async setPriority(classKey: SchemaItemKey, propertyName: string, priority: number) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setPriority(priority);
+  }
+
+  private async findDerivedClasses(mutableClass: MutableClass): Promise<Array<MutableClass>>{
+    const derivedClasses: Array<MutableClass> = [];
+    const schemaItems = this._schemaEditor.schemaContext.getSchemaItems();
+    let { value, done } = schemaItems.next();
+    while (!done) {
+      if (await value.is(mutableClass)) {
+        if (!mutableClass.key.matches(value.key)) {
+          derivedClasses.push(value);
+        }
+      }
+      ({ value, done } = schemaItems.next());
+    }
+
+    return derivedClasses;
+  }
+
+  /**
+   * Gets the property with the specified name from the class identified by the given key.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property to retrieve.
+   */
+  protected async getProperty<T extends MutablePropertyType>(classKey: SchemaItemKey, propertyName: string): Promise<T> {
+    let mutableClass: MutableClass;
+    try {
+      mutableClass = await this.getClass(classKey);
+    } catch (e: any) {
+      // TODO: Update error
+      throw new Error(e.message);
+    }
+
```

Comment:
Instead of the any type hint, I'd propose to use the BentleyError helper function here:
```ts
} catch (e: unknown) {
  throw new Error(BentleyError.getErrorMessage(e));
}

Reply:
done